assets/src/domain/eventEditor/ui/datetimes/hooks/useDatesListFilterBar.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useCallback } from 'react';
+import { __ } from '@wordpress/i18n';
+
+import { FilterBarSubscriptionCb } from '@appLayout/entityList/filterBar';
+import { DatetimesFilterStateManager } from '@edtrServices/filterState';
+import { DatetimesToShowControl, SortByControl, DisplayStartOrEndDateControl } from '../datesList/filterBar/controls';
+
+type DatesListFilterBarCallback = FilterBarSubscriptionCb<DatetimesFilterStateManager, 'dates-list'>;
+
+const useDatesListFilterBar = (): DatesListFilterBarCallback => {
+	return useCallback<DatesListFilterBarCallback>(({ listId }, { registerElement: registerFilterBarItem }) => {
+		// although this is not needed
+		if (listId !== 'dates-list') {
+			return;
+		}
+
+		registerFilterBarItem('datetimesToShow', ({ filterState }) => {
+			const { datetimesToShow, setDatetimesToShow } = filterState;
+			return (
+				<div className='ee-dates-datetimes-to-show-filter ee-filter-bar-filter'>
+					<DatetimesToShowControl datetimesToShow={datetimesToShow} setDatetimesToShow={setDatetimesToShow} />
+				</div>
+			);
+		});
```

Comment:
not that I want to do it now, but would be cool in the future maybe to also be able to register "decorators" for UI elements.
so instead of manually having to wrap all of these filter bar controls with:

```ts
<div className='ee-dates-datetimes-to-show-filter ee-filter-bar-filter'>
	// filter element
</div>
```

we would use a decorator to add that div wrapper, like:

```ts
const filterDecorator = ({filterElement, filterName}) => {
	<div className={`ee-filter-${filterName} ee-filter-bar-filter`}>{filterElement}</div>;
};
```